robot_controller:
  ros__parameters:
    ballance_controller:
      kp: [400.0, 500.0, 2500.0, 350.0, 500.0, 250.0 ]
      kd: [40.0, 50.0, 150.0, 20.0, 30.0, 15.0 ]
      s: [ 2.0, 4.0, 1.0, 40.0, 20.0, 18.0]
      w: 1e-3
      force_max: 1000.0
      force_min: -25.0
  
    joint_controller:
      kp: [200.0,300.0,300.0,
           200.0,300.0,300.0,
           200.0,300.0,300.0,
           200.0,300.0,300.0]

      kd: [10.0,20.0,20.0,
           10.0,20.0,20.0,
           10.0,20.0,20.0,
           10.0,20.0,20.0]

    leg_controller:
      kp: [200.0,500.0,2000.0,
           200.0,500.0,2000.0,
           200.0,500.0,2000.0,
           200.0,500.0,2000.0]
           
      kd: [10.0,15.0,35.0,
           10.0,15.0,35.0,
           10.0,15.0,35.0,
           10.0,15.0,35.0]

    nmpc_controller:
      n_steps: 10

geometric_nmpc_node:
  ros__parameters:
    geometric_nmpc:
      dt_MPC: 0.01
      n_steps: 10
      u: 0.6
      fmax: 1000.0
      fmin: 20.0
      vec_Q: [1.0, 100.0, 1.0, 0.01, 0.01, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0] #state cost rpyxyz
      vec_R: [1e-2, 1e-2, 1e-3] # force cost
      vec_P: [1e-6, 1e-6, 1e-6, 1e-6, 1e-6, 2e-1] #terminal cost